unstable/lens/struct-list.rkt

```diff
@@ -0,0 +1,59 @@
+#lang sweet-exp racket/base
+
+provide struct->list-lens list->struct-lens
+
+require racket/local
+        unstable/lens/isomorphism/base
+        for-syntax racket/base
+                   racket/list
+                   racket/struct-info
+                   syntax/parse
+module+ test
+  require lens/private/base/base
+          lens/private/test-util/test-lens
+          rackunit
+
+begin-for-syntax
+  (define-syntax-class struct-id
+    #:attributes (info constructor-id [accessor-id 1])
+    [pattern struct-id:id
+             #:attr v (syntax-local-value #'struct-id (λ () #f))
+             #:when (struct-info? (attribute v))
+             #:attr info (extract-struct-info (attribute v))
+             #:with descriptor-id:id (first (attribute info))
+             #:with constructor-id:id (syntax-property (second (attribute info))
+                                                       'disappeared-use
```

Comment:
So that it can draw check-syntax arrows like this:
![struct-lens-iso-arrow](https://cloud.githubusercontent.com/assets/6600123/11609763/a5d50720-9b5d-11e5-8494-00d8636bf99e.png)